udata/tests/api/test_datasets_api.py

```diff
@@ -990,113 +990,103 @@ def test_unfollow_dataset(self):
-                            qs={'q': 'test', 'size': '5'})
+        response = self.get(url_for('apiv2.suggest_formats'),
+                            qs={'q': 'km', 'size': '5'})
         self.assert200(response)
 
         self.assertLessEqual(len(response.json), 5)
         self.assertGreater(len(response.json), 1)
         # Shortest match first.
-        self.assertEqual(response.json[0]['text'], 'test')
+        self.assertEqual(response.json[0]['text'], 'kml')
 
         for suggestion in response.json:
             self.assertIn('text', suggestion)
-            self.assertIn('score', suggestion)
-            self.assertTrue(suggestion['text'].startswith('test'))
+            self.assertTrue(suggestion['text'].startswith('km'))
 
     def test_suggest_format_api_no_match(self):
         '''It should not provide format suggestion if no match'''
-        with self.autoindex():
-            DatasetFactory(resources=[
-                ResourceFactory(format=faker.word()) for _ in range(3)
-            ])
+        DatasetFactory(resources=[
+            ResourceFactory(format=faker.word()) for _ in range(3)
+        ])
 
-        response = self.get(url_for('api.suggest_formats'),
+        response = self.get(url_for('apiv2.suggest_formats'),
                             qs={'q': 'test', 'size': '5'})
         self.assert200(response)
         self.assertEqual(len(response.json), 0)
 
     def test_suggest_format_api_empty(self):
         '''It should not provide format suggestion if no data'''
         self.init_search()
-        response = self.get(url_for('api.suggest_formats'),
+        response = self.get(url_for('apiv2.suggest_formats'),
                             qs={'q': 'test', 'size': '5'})
         self.assert200(response)
         self.assertEqual(len(response.json), 0)
 
     def test_suggest_mime_api(self):
         '''It should suggest mime types'''
-        with self.autoindex():
-            DatasetFactory(resources=[
-                ResourceFactory(mime=f) for f in (
-                    faker.mime_type(category=None),
-                    faker.mime_type(category=None),
-                    'application/test',
-                    'application/test-1'
-                )
-            ])
+        DatasetFactory(resources=[
+            ResourceFactory(mime=f) for f in (
+                faker.mime_type(category=None),
+                faker.mime_type(category=None),
+                'application/json',
+                'application/json-1'
+            )
+        ])
 
-        response = self.get(url_for('api.suggest_mime'),
-                            qs={'q': 'test', 'size': '5'})
+        response = self.get(url_for('apiv2.suggest_mime'),
+                            qs={'q': 'js', 'size': '5'})
         self.assert200(response)
 
         self.assertLessEqual(len(response.json), 5)
-        self.assertGreater(len(response.json), 1)
         # Shortest match first.
```

Comment:
I don't think that in our case we sort by the shortest though? We're just being lucky here, no?